packages/mermaid/src/diagrams/xychart/parser/xychart.jison

```diff
@@ -0,0 +1,201 @@
+<acc_title>(?!\n|;|#)*[^\n]*              { this.popState(); return "acc_title_value"; }
+accDescr\s*":"\s*                         { this.begin("acc_descr");return 'acc_descr'; }
+<acc_descr>(?!\n|;|#)*[^\n]*              { this.popState(); return "acc_descr_value"; }
+accDescr\s*"{"\s*                         { this.begin("acc_descr_multiline");}
+<acc_descr_multiline>[\}]                 { this.popState(); }
+<acc_descr_multiline>[^\}]*               { return "acc_descr_multiline_value"; }
+
+"xychart-beta"		                        {return 'XYCHART';}
+("vertical"|"horizontal")                 {return 'CHART_ORIENTATION'}
+
+"x-axis"                                  { this.begin("axis_data"); return "X_AXIS"; }
+"y-axis"                                  { this.begin("axis_data"); return "Y_AXIS"; }
+<axis_data>[\[]                           { this.popState(); return 'SQUARE_BRACES_START'; }
+<axis_data>[+-]?\d+(?:\.\d+)?             { return 'NUMBER_WITH_DECIMAL'; }
+<axis_data>"-->"                          { return 'ARROW_DELIMITER'; }
+
+
+"line"                                    { this.begin("data"); return 'LINE'; }
+"bar"                                     { this.begin("data"); return 'BAR'; }
+<data>[\[]                                { this.popState(); this.begin("data_inner"); return 'SQUARE_BRACES_START'; }
+<data_inner>[+-]?\d+(?:\.\d+)?            { return 'NUMBER_WITH_DECIMAL';}
+<data_inner>[\]]                          { this.popState(); return 'SQUARE_BRACES_END'; }
+
+
+
+
+["][`]                                    { this.begin("md_string");}
+<md_string>[^`"]+                         { return "MD_STR";}
+<md_string>[`]["]                         { this.popState();}
+["]                                       this.begin("string");
+<string>["]                               this.popState();
+<string>[^"]*                             return "STR";
+
+
+[\[]                                      return 'SQUARE_BRACES_START'
+[\]]                                      return 'SQUARE_BRACES_END'
+[A-Za-z]+                                 return 'ALPHA';
+":"                                       return 'COLON';
+\+                                        return 'PLUS';
+","                                       return 'COMMA';
+"="                                       return 'EQUALS';
+\=                                        return 'EQUALS';
+"*"                                       return 'MULT';
+\#                                        return 'BRKT';
+[\_]                                      return 'UNDERSCORE';
+"."                                       return 'DOT';
+"&"                                       return 'AMP';
+\-                                        return 'MINUS';
+[0-9]+                                    return 'NUM';
+\s                                        /* skip */
+";"                                       return 'SEMI';
+[!"#$%&'*+,-.`?\\_/]                      return 'PUNCTUATION';
```

Comment:
It seems, than PUNCTUATION unites the majority of previous symbols. But it will never be recognized because of the lexems order. There are 2 options, I think:
* either remove `PUNCTUATION`, and remove it from `alphaNumToken` as well, and leave all of those separate tokens, thus we would get much nicer and clearer parsing error message if any
* or leave PUNCTUATION only, since it already includes everything else

Reply:
That make sense, Removed the PUNCTUATION